wurst/event/OnUnitEnterLeave.wurst

```diff
@@ -36,24 +36,24 @@ constant ABILITY_ID = compiletime(ABIL_ID_GEN.next())
 
 constant eventTrigger = CreateTrigger()
 unit tempUnit = null
-group preplacedUnits
+let preplacedUnits = CreateGroup()
 constant trigger preplacedUnitsCall = CreateTrigger()..addAction(function initPreplacedUnits)
 
 function prepareUnit(unit u)
 	u..addAbility(ABILITY_ID)..makeAbilityPermanent(ABILITY_ID, true)
 
 function initPreplacedUnits()
-	for count=1 to UNITS_PER_CALL
-		let u = FirstOfGroup(preplacedUnits)
-		if u == null
+	var count = UNITS_PER_CALL
+
+	for u from preplacedUnits
+		prepareUnit(u)
+		eventTrigger.evaluate()
+		count--
+		if count <= 0
 			break
-		else
-			preplacedUnits.removeUnit(u) 
-			prepareUnit(u)
-			eventTrigger.evaluate()
 
-	if FirstOfGroup(preplacedUnits) != null
-		preplacedUnitsCall.execute()
+	if preplacedUnits.hasNext()
+		preplacedUnitsCall.execute() // processes at most UNITS_PER_CALL units per thread
```

Comment:
I also used that word "thread", but it's not really correct.

Dracolich showed that this op-limit is a single value incrementing, so maybe `processes at most UNITS_PER_CALL units before resetting the op-limit.`